### PR TITLE
[GraphBolt] Add `indexes` to MiniBatch.

### PR DESCRIPTION
**Pull request by yxy235**
## Description

Add `indexes` to MiniBatch, which indicates to which query a seed node / node pair belongs.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/minibatch.py [hunk 1]
```diff
@@ -80,6 +80,18 @@ class MiniBatch:
           or other graph components depending on the specific context.
     """
 
+    query_index: Union[torch.Tensor, Dict[str, torch.Tensor]] = None
+    """
+    Group ids associated with seed nodes / node pairs in the graph, which
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:85]:
```suggestion
    Query indexes associated with seed nodes / node pairs in the graph, which
```

File: python/dgl/graphbolt/minibatch.py [hunk 2]
```diff
@@ -80,6 +80,18 @@ class MiniBatch:
...
+    indicates to which query a seed node / node pair belongs.
+    - If `query_index` is a tensor: It indicates the graph is homogeneous. The
+      value should be corresponding query to given 'seed_nodes' or
+      'node_pairs'.
+    - If `query_index` is a dictionary: It indicates the graph is
+      heterogeneous. The keys should be node or edge type and the value should
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:91]: Will the key be the same across different node / edge type?

Comment (peizhou001) [reply python/dgl/graphbolt/minibatch.py:91]: Use `query` or `indexes` or `group_id`

File: python/dgl/graphbolt/minibatch.py [hunk 3]
```diff
@@ -80,6 +80,20 @@ class MiniBatch:
           or other graph components depending on the specific context.
     """
 
+    query_index: Union[torch.Tensor, Dict[str, torch.Tensor]] = None
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:83]:
```suggestion
    indexes: Union[torch.Tensor, Dict[str, torch.Tensor]] = None
```
Align with torch's definition:
https://lightning.ai/docs/torchmetrics/stable/retrieval/mrr.html

File: python/dgl/graphbolt/minibatch.py [hunk 4]
```diff
@@ -80,6 +80,20 @@ class MiniBatch:
...
+    Indexes associated with seed nodes / node pairs in the graph, which
+    indicates to which query a seed node / node pair belongs.
+    - If `query_index` is a tensor: It indicates the graph is homogeneous. The
+      value should be corresponding query to given 'seed_nodes' or
+      'node_pairs'.
+    - If `query_index` is a dictionary: It indicates the graph is
+      heterogeneous. The keys should be node or edge type and the value should
+      be corresponding query to given 'seed_nodes' or 'node_pairs'. The
+      starting value of each query ids is 0, and the query ids between
+      different keys do not affect each other.
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:94]:
```suggestion
      be corresponding query to given 'seed_nodes' or 'node_pairs'. For
      each key, indexes are consecutive integers starting from zero.
```

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**